### PR TITLE
Change language statistics to save size instead of percentage

### LINKED ISSUE #10567: Hide SVG (or more) from language stats panel.
- Gitea version (or commit ref): 1.12.0+dev-412-g4160bd6ef

## Description

It would be cool if you can configure which languages to hide in the language stats panel.  
If this is to much work, we could consider hiding svg by default, because that are mainly assets.

## Screenshots

Screenshot_20200302_113201

### PR DESCRIPTION
**Pull request by lafriks**
Closes #11672

Fixes #10567

Co-Authored-By: Cirno the Strongest <1447794+CirnoT@users.noreply.github.com>

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Review by CirnoT — CHANGES_REQUESTED** [review]

File: models/repo_language_stats.go [hunk 1]
```diff
@@ -34,12 +51,41 @@ func (stats LanguageStatList) loadAttributes() {
 	}
 }
 
+func (stats LanguageStatList) getLanguagePercentages() map[string]float32 {
+	langPerc := make(map[string]float32)
+	var otherPerc float32 = 100
+	var total int64
+	for _, stat := range stats {
```
Comment (CirnoT) [inline models/repo_language_stats.go:60]: There is no reason to range every language in `stat`, you can just range every language in `specialLanguages` and perform `delete` (which is no-op if doesn't exist) on `stats` with that key.

Comment (CirnoT) [reply models/repo_language_stats.go:60]: That also means you'll be able to move `specialLanguages` from map to slice.

Comment (lafriks) [reply models/repo_language_stats.go:60]: I still need to loop to calculate total size & percentages in second loop

**Review by CirnoT — APPROVED** [review]

**Comment by silverwind** [comment]
How does this fix the case of `yarn.lock` exactly?

**Comment by lafriks** [comment]
@silverwind what do you mean

**Comment by silverwind** [comment]
You linked https://github.com/go-gitea/gitea/pull/11672 but I fail to see how it's related.

**Comment by CirnoT** [comment]
@silverwind It includes fix from #11672 along with necessary changes, so that there is no need to resolve merge conflicts for either of PRs.

**Comment by silverwind** [comment]
Ah, missed that. Carry on.

**Comment by zeripath** [comment]
@lafriks needs an update

**Comment by lafriks** [comment]
@zeripath fixed some edge cases with special languages, should pass tests now

**Comment by lafriks** [comment]
make lg-tm work
